src/D2L.CodeStyle.Analyzers/Pinning/PinnedAttributeAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.Pinning {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	internal sealed class PinnedAttributeAnalyzer : DiagnosticAnalyzer {
+
+		private static readonly TypeKind[] TypeKindsSupported = new[] { TypeKind.Class, TypeKind.Interface };

```

Comment:
The attribute also supports being applied to `struct`. Is this check necessary given that we've already cast to `INamedTypeSymbol` at the point we check this? Can we just trust the `AttributeUsage` definition?

Reply:
We also support `record` types. That might be a class/struct because I don't see a record enum